### ISSUE TITLE
`flit` and `flit_core` can be out of sync

### ISSUE THREAD

**Issue by pmeier** [takluyver/flit#529]
For example 

```shell
$ pip install flit==3.6.0
[...]
$ python -c "import flit_core; print(flit_core.__version__)"
3.7.0
```

This was detected, because the 3.7.0 release broke some build workflows (still investigating what exactly happened, see omnilib/ufmt#56) although `flit==3.6.0` is pinned.

**Comment by takluyver** [comment]
Sorry about that.

I'm not sure that 'out of sync' is really the problem, though - I think the changes in #499 may simply have broken it for you.

I don't think your pinning in `requirements-dev.txt` affects this at all: [pre-commit uses](https://pre-commit.com/#python) `pip install .` which will look at `pyproject.toml`, and set up an isolated build environment with `flit_core` according to the `[build-system]` table.

**Comment by pmeier** [comment]
> I don't think your pinning in `requirements-dev.txt` affects this at all: [pre-commit uses](https://pre-commit.com/#python) `pip install .` which will look at `pyproject.toml`, and set up an isolated build environment with `flit_core` according to the `[build-system]` table.

True. The pinning would only solve this for the development environment.

**Comment by takluyver** [comment]
The breakage you're seeing has just been reported as #530. I'll leave this one open for the lower-priority discussion about what the dependency relationship between flit and flit_core should be.
